core/compiler.cpp

```diff
@@ -307,7 +310,14 @@ torch::jit::Module CompileGraph(const torch::jit::Module& mod, CompileSpec cfg)
       // Infer the type of an input from the weights of the calculation
       auto first_use_types = ir::get_block_first_calc_dtypes_opt_collection(g->block());
 
-      MapInputsAndDetermineDTypes(cfg, g, static_params, first_use_types);
+      // Extract map of IValue to DType
+      auto type_map = MapInputsAndDetermineDTypes(cfg, g, static_params, first_use_types);
+
+      // Use dtype map to autocast inputs to the correct type
+      if (cfg.partitioning_info.enabled && cfg.partitioning_info.truncate_long_and_double) {
```

Comment:
An alternative to having the `truncate_long_and_double` flag control this feature is to add a new flag solely for `autocast` in case users want to have internal tensors truncated, but don't want the overhead of an additional engine to truncate inputs.